### ISSUE TITLE
Feature Request: Add hover inspection Image glyphs

### ISSUE THREAD

**Issue by ofirsa** [bokeh/bokeh#3886]
Hi, I hope this the right place to ask my question.
is it possible to implement the Hover Tool on image in such a way that when I moving my mouse on the image pixels, I get and information on the pixel below the mouse, like the pixel coordinates and pixel value.
I would be glad to get an example.
Thanks
Saar

**Comment by birdsarah** [comment]
Hi @ofirsa, our issue tracker is becoming too large to manage easily. We have to reserve it for feature requests and bug reports only, to have any hope of taming it. This kind of general support question should go to the general public mailing list:

https://groups.google.com/a/continuum.io/forum/#!forum/bokeh

**Comment by jbednar** [comment]
Sorry to revisit this old issue, but as far as I can tell it is still valid.  Sarah closed it as a support question, but to me it sounds like either a feature request (can images please support hover information like other glyphs do) or a bug report (images don't support hover information like they should), as requested.

In either case, what's the answer?  Is there some fundamental reason why Image shouldn't support hovering to reveal the pixel coordinates and the value of the image at that point?  We will need this in our work, and can start working on a PR if that's appropriate.

**Comment by jbednar** [comment]
There's some relevant discussion on the Google Group, but nothing that indicates to me that this feature request is somehow invalid or not feasible:

https://groups.google.com/a/continuum.io/forum/#!searchin/bokeh/hover$20tool$20image|sort:relevance/bokeh/W-n34ElNlRo/O7_eut-5DwAJ

https://groups.google.com/a/continuum.io/forum/#!searchin/bokeh/Hover$20image/bokeh/fOlnivwLwvw/miZDTdA8CQAJ

**Comment by bryevdv** [comment]
Not sure why this was closed, I thought there was another issue for image hit-testing but I can't find it offhand. Doing this will required setting a "2d" selection, i.e, will want to know which `i, j` pixel in the image was hit. That's not common, but there should already be kind-of similar example code for `MultiLine`. There's also the question of what to do when the image is big and a single pixel on the screen spans more than one pixel in the image (e.g. 1000x1000 image in a 500x500 canvas). 

Certainly possible in principle, but mostly just needs someone to spend some time thinking about cases like the above, and defining and documenting the exact behaviors. Will need tests too.

**Comment by bryevdv** [comment]
To be clear: would be a welcome addition, just needs a little upfront thought and design.

**Comment by jbednar** [comment]
Sounds good.  Happy for Jean-Luc to take a stab at it?

**Comment by bryevdv** [comment]
For sure.

**Comment by jkquijas** [comment]
Hi! Any update on this feature request/bug report? This functionality can be very useful.

**Comment by jbednar** [comment]
It's still on @jlstevens's to-do list, and is expected to be addressed soon.

**Comment by jlstevens** [comment]
Yes, I am eager to tackle this although I am currently waiting on the refactor in #7261 to be completed first.

**Comment by jkquijas** [comment]
Awesome! Thanks!!
